File: planningMaintenance/src/main/java/net/elghz/planningmaintenance/model/Intervention.java
package net.elghz.planningmaintenance.model;

import lombok.Data;
import net.elghz.planningmaintenance.enumeration.statusIntervention;

import java.time.LocalTime;
import java.util.Date;
@Data
public class Intervention {
    private Long id;
    private String name;
    private Date dateDebut;

    private Date dateFin;
    private LocalTime heureDebut;
    //private PrioriteEnum priorite;

    private ResponsableMaint responsable;
    private String description;
    private Long id_Planning;

   // private Planning planning;
   // private TypeIntervention type;
    private statusIntervention status;
    private String id_Techn;

    private ResponsableMaint technicien;

    private Long id_Equipement;
   // private Equipement equipement;
}

File: frontEnd-Angular/src/app/models/intervention.ts
import {Technicein} from "./technicein";
import {Equipement} from "./equipement";
import {User} from "./user";
import {PlanningMaintenanceDTO} from "./PlanningMaintenanceDTO";

export interface Intervention{
    id:number;
    name:string;
    dateDebut:Date

    dateFin:Date;
  priorite:String

    description:String;

    status:String;
    technicien:User
planning:PlanningMaintenanceDTO
  type:string,
  equipement:Equipement,
  heureDebut: string
  responsable:User;

}
